entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/mapper/ViewMapper.java

```diff
@@ -0,0 +1,320 @@
+                }
+
+                MapInstantiator<?, ?> mapInstantiator = ((AbstractAttribute<?, ?>) targetAttribute).getMapInstantiator();
+                return new MapMappingAccessor(Accessors.forViewAttribute(null, sourceAttribute, true), needsDirtyTracker, mapInstantiator, keyMapper, valueMapper);
+            } else {
+                CollectionInstantiator collectionInstantiator = ((AbstractAttribute<?, ?>) targetAttribute).getCollectionInstantiator();
+                return new CollectionMappingAccessor(Accessors.forViewAttribute(null, sourceAttribute, true), needsDirtyTracker, collectionInstantiator, valueMapper);
+            }
+        } else if (targetAttribute.isSubview()) {
+            ViewMapper<Object, Object> mapper = createViewMapper(((SingularAttribute<?, ?>) sourceAttribute).getType(), ((SingularAttribute<?, ?>) targetAttribute).getType(), ignoreMissing, proxyFactory);
+            return new AttributeMappingAccessor(Accessors.forViewAttribute(null, sourceAttribute, true), mapper);
+        } else if (targetAttribute.getJavaType() != sourceAttribute.getJavaType()) {
+            throw inconvertible("Attribute '" + targetAttribute.getName() + "' from target type has a different type than in source type!", sourceType, targetType);
+        } else {
+            return Accessors.forViewAttribute(null, sourceAttribute, true);
+        }
+    }
+
+    private ViewMapper<Object, Object> createViewMapper(Type<?> source, Type<?> target, boolean ignoreMissing, ProxyFactory proxyFactory) {
+        ManagedViewType<Object> sourceType = (ManagedViewType<Object>) source;
+        ManagedViewType<Object> targetType = (ManagedViewType<Object>) target;
+        return new ViewMapper<>(sourceType, targetType, ignoreMissing, proxyFactory);
+    }
+
+    private RuntimeException inconvertible(String reason, ManagedViewType<S> sourceType, ManagedViewType<T> targetType) {
+        return new IllegalArgumentException("Can't convert from '" + sourceType.getJavaType().getName() + "' to '" + targetType.getJavaType().getName() + "'! " + reason);
+    }
+
+    public T map(S source) {
+        Object[] tuple = new Object[sourceAccessors.length];
+        for (int i = 0; i < sourceAccessors.length; i++) {
+            if (sourceAccessors[i] != null) {
+                tuple[i] = sourceAccessors[i].getValue(source);
+            }
+        }
+        return objectInstantiator.newInstance(tuple);
+    }
+
+    public static class Key<S, T> {
+        private final ManagedViewType<S> sourceType;
+        private final ManagedViewType<T> targetType;
+        private final boolean ignoreMissing;
+
+        public Key(ManagedViewType<S> sourceType, ManagedViewType<T> targetType, boolean ignoreMissing) {
+            this.sourceType = sourceType;
+            this.targetType = targetType;
+            this.ignoreMissing = ignoreMissing;
+        }
+
+        public ViewMapper<S, T> createMapper(ProxyFactory proxyFactory) {
```

Comment:
Why is the `Key` able to create a `ViewMapper`? Shouldn't the `Key` be merely a value holder to facilitate a compound map key? Looks to me like violating the single responsibility principle for no reason.